test/unit/helpers/foundation.js

```diff
@@ -31,13 +31,89 @@ import td from 'testdouble';
 // Every foundation test suite include this verification.
 export function verifyDefaultAdapter(FoundationClass, expectedMethods) {
   const {defaultAdapter} = FoundationClass;
-  const methods = Object.keys(defaultAdapter).filter((k) => typeof defaultAdapter[k] === 'function');
+  const actualMethods = Object.keys(defaultAdapter).filter((k) => typeof defaultAdapter[k] === 'function');
+
+  assert.equal(actualMethods.length, Object.keys(defaultAdapter).length, 'Every adapter key must be a function');
 
-  assert.equal(methods.length, Object.keys(defaultAdapter).length, 'Every adapter key must be a function');
   // Test for equality without requiring that the array be in a specific order
-  assert.deepEqual(methods.slice().sort(), expectedMethods.slice().sort());
+  const actualArray = actualMethods.slice().sort();
+  const expectedArray = expectedMethods.slice().sort();
+  assert.deepEqual(actualArray, expectedArray, getUnequalArrayMessage(actualArray, expectedArray));
+
   // Test default methods
-  methods.forEach((m) => assert.doesNotThrow(defaultAdapter[m]));
+  actualMethods.forEach((m) => assert.doesNotThrow(defaultAdapter[m]));
+}
+
+/**
+ * @param {!Array<string>} actualArray
+ * @param {!Array<string>} expectedArray
+ * @return {string}
+ */
+function getUnequalArrayMessage(actualArray, expectedArray) {
+  /**
+   * @param {!Array<string>} values
+   * @param {string} singularName
+   * @return {string}
+   */
+  const format = (values, singularName) => {
+    const count = values.length;
+    if (count === 0) {
+      return '';
+    }
+    const plural = count === 1 ? '' : 's';
+    const str = values.join(', ');
+    return `${count} ${singularName}${plural}: ${str}`;
+  };
+
+  /**
+   * @param {!Set<string>} actualSet
+   * @param {!Set<string>} expectedSet
+   */
+  const getAddedStr = (actualSet, expectedSet) => {
+    const addedArray = [];
+    [...actualSet].forEach((val) => {
+      if (!expectedSet.has(val)) {
```

Comment:
`has()` is not supported in IE & Safari. This would throw error if we run our unit test in IE on travis unless babel transpiles this.